docs/appendix-A.rst

```diff
@@ -0,0 +1,1783 @@
+   :end-before: end getConstStringRefPure
+
+YAML:
+
+.. code-block:: yaml
+
+    - decl: const string& getConstStringRefPure()
+
+The C wrapper:
+
+.. literalinclude:: ../regression/reference/strings/wrapstrings.cpp
+   :language: c
+   :start-after: start STR_get_const_string_ref_pure_bufferify
+   :end-before: end STR_get_const_string_ref_pure_bufferify
+
+The native C wrapper:
+
+.. literalinclude:: ../regression/reference/strings/wrapstrings.cpp
+   :language: c
+   :start-after: start STR_get_const_string_ref_pure
+   :end-before: end STR_get_const_string_ref_pure
+
+Calls C via the interface:
+
+.. literalinclude:: ../regression/reference/strings/wrapfstrings.f
+   :language: fortran
+   :start-after: start c_get_const_string_ref_pure_bufferify
+   :end-before: end c_get_const_string_ref_pure_bufferify
+   :dedent: 8
+
+The Fortran wrapper:
+
+.. literalinclude:: ../regression/reference/strings/wrapfstrings.f
+   :language: fortran
+   :start-after: start get_const_string_ref_pure
+   :end-before: end get_const_string_ref_pure
+   :dedent: 4
+
+Fortran usage:
+
+.. code-block:: fortran
+
+    str = 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
+    str = get_const_string_ref_pure()
+    call assert_true( str == static_str, "getConstStringRefPure")
+
+
+std::vector
+-----------
+
+.. ############################################################
+
+.. _example_vector_sum:
+
+vector_sum
+^^^^^^^^^^
+
+.. literalinclude:: ../regression/run/vectors/vectors.cpp
+   :language: c
+   :start-after: start vector_sum
+   :end-before: end vector_sum
+
+YAML:
+
+.. code-block:: yaml
+
+    - decl: int vector_sum(const std::vector<int> &arg)
+
+``intent(in)`` is implied for the *vector_sum* argument since it is
+``const``.  The Fortran wrapper passes the array and the size to C.
+
+The C wrapper:
+
+.. literalinclude:: ../regression/reference/vectors/wrapvectors.cpp
+   :language: c
+   :start-after: start VEC_vector_sum_bufferify
+   :end-before: end VEC_vector_sum_bufferify
+
+Calls C via the interface:
+
+.. literalinclude:: ../regression/reference/vectors/wrapfvectors.f
+   :language: fortran
+   :start-after: start c_vector_sum_bufferify
+   :end-before: end c_vector_sum_bufferify
+   :dedent: 8
+
+The Fortran wrapper:
+
+.. literalinclude:: ../regression/reference/vectors/wrapfvectors.f
+   :language: fortran
+   :start-after: start vector_sum
+   :end-before: end vector_sum
+   :dedent: 4
+
+Fortran usage:
+
+.. code-block:: fortran
+
+    integer(C_INT) intv(5)
+    intv = [1,2,3,4,5]
+    irv = vector_sum(intv)
+    call assert_true(irv .eq. 15)
+
+.. ############################################################
+
+.. _example_vector_iota_out:
+
+vector_iota_out
+^^^^^^^^^^^^^^^
+
+.. literalinclude:: ../regression/run/vectors/vectors.cpp
+   :language: c
+   :start-after: start vector_iota_out
+   :end-before: end vector_iota_out
+
+YAML:
+
+.. code-block:: yaml
+
+    - decl: void vector_iota_out(std::vector<int> &arg+intent(out))
+
+The C wrapper:
```

Comment:
Might be worthwhile adding a note about how shroud arranges for this not to leak.

```cpp
void VEC_vector_iota_out_bufferify(VEC_SHROUD_array *Darg)
{
// splicer begin function.vector_iota_out_bufferify
    std::vector<int> *SH_arg = new std::vector<int>;
    vector_iota_out(*SH_arg);
    Darg->cxx.addr  = static_cast<void *>(SH_arg);
    Darg->cxx.idtor = 1;
    Darg->addr.cvoidp = SH_arg->empty() ? NULL : &SH_arg->front();
    Darg->len = sizeof(int);
    Darg->size = SH_arg->size();
    return;
// splicer end function.vector_iota_out_bufferify
}
```

Reply:
Done